### PR TITLE
configure-nilrt-snac: Configure tmux for all logins and set timeout to lock

### PR DESCRIPTION
**Pull request by texasaggie97**
### Summary of Changes

Install and configure tmux to be the default shell for all logins, local and ssh, then set a timeout to lock the session after 15 minutes of inactivity.

### Justification

[User Story 2816950](https://dev.azure.com/ni/DevCentral/_workitems/edit/2816950): [SNAC Auth] Configure tmux to be default shell as part of SNAC configuration tool

### Testing

Copied new function to test system and verified that tmux is properly set up afterwards

### Procedure

* [X] I certify that the contents of this pull request complies with the [Developer Certificate of Origin](https://github.com/ni/nilrt-snac/blob/master/docs/CONTRIBUTING.md).

### CODE REVIEW

File: src/configure-nilrt-snac [hunk 1]
```diff
@@ -141,6 +141,29 @@ remove_niauth() {
 }
 
 
+configure_tmux() {
+	log INFO Configuring tmux to be the default shell for both SSH and local
```
Comment (dmondrik) [inline src/configure-nilrt-snac:145]: Consider: This is more verbose than above, where we just say we're configuring opkg or NTP but don't say how. Not sure what level of detail we ought to provide, but I'd expect minimal would suffice.

Comment (texasaggie97) [reply src/configure-nilrt-snac:145]: If you look at the `configure_wireguard()` function, while the log lines are short, there are multiple of them explaining what is happening as it configures it. I can shorten this to a single short log, break it into multiple log lines line `configure_wireguard()`, or leave it as is, whatever is decided.

File: src/configure-nilrt-snac [hunk 2]
```diff
@@ -141,6 +141,29 @@ remove_niauth() {
...
+	log INFO    logins, with a timeout...
+	opkg update >/dev/null && opkg install tmux
+
+	cat <<EOF >/etc/tmux.conf
+# NILRT SNAC configuration tmux.conf. Do not hand-edit.
+
+set -g lock-command vlock
```
Comment (amstewart) [inline src/configure-nilrt-snac:152]: We can upstream this `lock-command` config line to meta-nilrt.

That would mean adding a new directory [here](https://github.com/ni/meta-nilrt/tree/nilrt/master/next/recipes-extended) with files like

`files/tmux.conf`
```
set -g lock-command vlock
bind X lock-session
```

`tmux_3.%.bbappend`
```
FILESEXTRAPATHS:prepend := "${THISDIR}/files:"

SRC_URI += "\
    file://tmux.conf \
"

do_install:append() {
    install -d ${D}${sysconfdir}
    install -m 644 ${WORKDIR}/tmux.conf ${D}${sysconfdir}/tmux.conf
}

RDEPENDS:${PN}:append = " vlock"
```

**Review by amstewart — CHANGES_REQUESTED** [review]

File: src/configure-nilrt-snac [hunk 3]
```diff
@@ -141,6 +141,29 @@ remove_niauth() {
...
+set -g lock-after-time 900
+EOF
```
[the remark this answers is not in the document]
Comment (texasaggie97) [reply src/configure-nilrt-snac]: Changed

File: src/configure-nilrt-snac (hunk 2 above)
Comment (texasaggie97) [reply src/configure-nilrt-snac]: Done. See ni/meta-nilrt#718

File: src/configure-nilrt-snac [hunk 4]
```diff
@@ -188,6 +188,30 @@ EOF
...
+	opkg update >/dev/null && opkg install tmux
+
+	log INFO Create timeout configuration for tmux and source it
+	mkdir -p /usr/share/nilrt-snac
+	cat <<EOF >/usr/share/nilrt-snac/tmux-snac.conf
+# NILRT SNAC configuration tmux-snac.conf. Do not hand-edit.
+
+set -g lock-after-time 900
+EOF
+	echo "source-file /usr/share/nilrt-snac/tmux-snac.conf" >>/etc/tmux.conf
```
Comment (amstewart) [inline src/configure-nilrt-snac:202]: Thanks for making this change, I'll just note that we could get rid of this line if you prefer my [design suggestion](https://github.com/ni/meta-nilrt/pull/718#discussion_r1722189535) over in meta-nilrt.

File: src/configure-nilrt-snac [hunk 5]
```diff
@@ -188,6 +188,30 @@ EOF
...
+	log INFO Install tmux
+	opkg update >/dev/null && opkg install tmux
+
+	log INFO Create timeout configuration for tmux and source it
+	mkdir -p /usr/share/nilrt-snac
+	cat <<EOF >/usr/share/nilrt-snac/tmux-snac.conf
+# NILRT SNAC configuration tmux-snac.conf. Do not hand-edit.
+
+set -g lock-after-time 900
+EOF
```
Comment (amstewart) [inline src/configure-nilrt-snac:201]: FYI: You can use `<<-EOF` (with a minus-sign) to direct bash to ignore leading whitespace in your heredoc. Then you can indent the heredoc without the indents carrying through.

**Review by amstewart — COMMENTED** [review]
Changes look good. I'll just hold off on the merge until I know if you want to change the tmux.conf source logic.

**Comment by texasaggie97** [comment]
Starting over with a new PR with better commits. All comments here addressed. See #10
